src/main/java/doore/team/application/TeamCommandService.java

```diff
@@ -53,16 +63,45 @@ public void updateTeamImage(final Long teamId, final MultipartFile file) {
     }
 
     public void deleteTeam(final Long teamId) {
-        final Team team = findTeamById(teamId);
+        final Team team = validateExistTeam(teamId);
         teamRepository.delete(team);
         if (team.hasImage()) {
             s3ImageFileService.deleteFile(team.getImageUrl());
         }
         // TODO: 2/2/24 팀이 삭제될 시 연관된 스터디와, 커리큘럼도 삭제
     }
 
-    private Team findTeamById(final Long teamId) {
+    private Team validateExistTeam(final Long teamId) {
         return teamRepository.findById(teamId)
                 .orElseThrow(() -> new TeamException(NOT_FOUND_TEAM));
     }
+
+    public TeamInviteCodeResponse generateTeamInviteCode(final Long teamId) {
+        validateExistTeam(teamId);
+
+        final Optional<String> link = redisUtil.getData(INVITE_LINK_PREFIX.formatted(teamId), String.class);
+        if (link.isEmpty()) {
+            final String randomCode = RandomUtil.generateRandomCode('0', 'z', 10);
+            redisUtil.setDataExpire(INVITE_LINK_PREFIX.formatted(teamId), randomCode, RedisUtil.toTomorrow());
+            return new TeamInviteCodeResponse(randomCode);
+        }
+        return new TeamInviteCodeResponse(link.get());
```

Comment:
```suggestion
        String inviteKey = INVITE_LINK_PREFIX.formatted(teamId);
        return redisUtil.getData(inviteKey, String.class)
                       .map(TeamInviteCodeResponse::new)
                       .orElseGet(() -> {
                String randomCode = RandomUtil.generateRandomCode('0', 'z', 10);
                redisUtil.setDataExpire(inviteKey, randomCode, RedisUtil.toTomorrow());
                return new TeamInviteCodeResponse(randomCode);
            });
```

이렇게도 작성할 수 있을 것 같습니다~ 코틀린을 공부하고 오니 온 세상이 유사 스트림 처럼 처리할 수 없을까 생각이 드네요.. 작성하고 보니까 @shkisme 님이 작성해 주신 코드가 오히려 가독성이 더 좋을 수 있겠다고 생각이 드네요~

Reply:
오우 ㅋㅋㅋㅋ 스트림 무림 고수가 되셨군요.

제 생각엔 `Redis`에서 찾을 수 없을때 수행되는 `orElseGet`에 중요한 로직이 들어있는건 가독성이 조금 떨어진다고 생각하여 그대로 두겠습니다!! 좋은 피드백 감사합니다 😊